tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+                                glue_nested_table_name.replace("_", "=", 1)
+                            ) + "/"
+                        }
+                    ]
+                },
+                SchemaChangePolicy={
+                    "DeleteBehavior": "LOG",
+                    "UpdateBehavior": "LOG"
+                },
+                RecrawlPolicy={
+                    "RecrawlBehavior": "CRAWL_NEW_FOLDERS_ONLY"
+                },
+                Configuration=json.dumps({
+                    "Version":1.0,
+                    "CrawlerOutput": {
+                        "Partitions": {
+                            "AddOrUpdateBehavior":"InheritFromTable"
+                        }
+                    },
+                    "Grouping": {
+                        "TableGroupingPolicy":"CombineCompatibleSchemas"
+                    }
+                })
+        )
+        glue_client.start_crawler(Name=crawler_name)
+        response = {"Crawler": {}}
+        for i in range(60): # wait up to 10 minutes for crawler to finish
+            # This should take about 5 minutes
+            response = glue_client.get_crawler(Name=crawler_name)
+            if (
+                    "LastCrawl" in response["Crawler"]
+                    and "Status" in response["Crawler"]["LastCrawl"]
+                    and response["Crawler"]["LastCrawl"]["Status"] == "SUCCEEDED"
+               ):
+                break
+            else:
+                time.sleep(10)
+        yield glue_crawler
+        glue_client.delete_crawler(Name=crawler_name)
+
+    @pytest.fixture()
+    def json_s3_objects(self, datadir, artifact_bucket, namespace):
+        s3_client = boto3.client("s3")
+        dataset_prefix = os.path.join(namespace, "tests")
+        object_keys = []
+        for dirpath, _, filenames in os.walk(datadir):
+            if len(filenames) > 0:
+                object_prefix = os.path.join(
+                        dataset_prefix,
+                        os.path.join(*dirpath.split("/")[-6:]))
+                for filename in filenames:
+                    object_key = os.path.join(object_prefix, filename)
+                    s3_client.upload_file(
+                            Filename=os.path.join(dirpath, filename),
+                            Bucket=artifact_bucket,
+                            Key=object_key)
+                    object_keys.append(object_key)
+        return object_keys
+
+    @pytest.fixture(scope="class")
+    def glue_context(self):
+        glue_context = GlueContext(SparkSession.builder.getOrCreate())
+        return glue_context
+
+    def test_upload_test_data(self, glue_test_data):
+        """
+        Upload test data to S3.
+        """
+        pass
+
+    def test_run_glue_crawler(self, glue_crawler):
+        """
+        Create and run Glue crawler.
+        """
+        pass
+
+    def test_get_table(self, glue_database_name, glue_flat_table_name,
```

Comment:
Could use something like `@pytest.mark.parametrize` to separate out these
two tests and use the `ids` parameter to title what each test is testing so it will show up when you run pytest: https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
I think it's possible but since we are testing two different things (that both `table.count` is correct and `len(table.schema().fields` is correct) it would be a bit hacky and I'd rather use this straightforward, redundant way of testing then use an abstracted but hacky way.

Reply:
Yeah I can see that it would look very hacky... Maybe we could separate it out into two different tests then and update the test function names? I'm just thinking of when this script grows and we add more tests or in the future if we haven't touched this in a while, and we get a test report and there are failing tests, it would be good to have the nested table and regular table be two separate tests here so they will be concurrently run instead of now where if the regular table test fails, we won't know if nested table test would fail or pass.

Reply:
I split these up.